### PR TITLE
Add ElementHandle::accessible_role()

### PR DESCRIPTION
**Pull request by tronical**
This allows querying for elements by their role/type.

### CODE REVIEW

File: internal/backends/testing/lib.rs [hunk 1]
```diff
@@ -37,3 +37,5 @@ pub fn init_integration_test() {
     i_slint_core::platform::set_platform(Box::new(testing_backend::TestingBackend::new()))
         .expect("platform already initialized");
 }
+
+pub use i_slint_core::items::AccessibleRole;
```
Comment (ogoffart) [inline internal/backends/testing/lib.rs:41]: If we're exporting this, we should at least mark it as #[non_exhaustive]

Comment (tronical) [reply internal/backends/testing/lib.rs:41]: Good point!!

File: internal/interpreter/dynamic_item_tree.rs [hunk 2]
```diff
@@ -1915,6 +1915,8 @@ extern "C" fn item_geometry(component: ItemTreeRefPin, item_index: u32) -> Logic
     }
 }
 
+// silent the warning despite `Clipboard` is a `#[non_exhaustive]` enum from another crate.
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/interpreter/dynamic_item_tree.rs:1918]: Oops!

**Comment by tronical** [comment]
> And should there be an equivalent C++ API?

Yes.

**Pull request merged**